Sources/BottomSheetPresentationController.swift

```diff
@@ -39,6 +39,54 @@ final class BottomSheetPresentationController: UIPresentationController {
     // MARK: - Transition life cycle
 
     override func presentationTransitionWillBegin() {
+        guard transitionState == .presenting else { return }
+        createBottomSheetView()
+    }
+
+    override func presentationTransitionDidEnd(_ completed: Bool) {
+        guard transitionState == nil else { return }
+        guard let containerView = containerView else { return }
+
+        createBottomSheetView()
+
+        bottomSheetView?.present(
+            in: containerView,
+            targetIndex: startTargetIndex,
+            animated: false
+        )
+    }
+
+    override var frameOfPresentedViewInContainerView: CGRect {
+        guard let presentedView = presentedView else { return .zero }
+        guard let containerView = containerView else { return .zero }
+
+        let offset = BottomSheetCalculator.offset(
+            for: presentedView,
+            in: containerView,
+            height: targetHeights[startTargetIndex]
+        )
+
+        let handleHeight: CGFloat = 20
+        let expectedHeight: CGFloat
+
+        if targetHeights[startTargetIndex] == .bottomSheetAutomatic {
```

Comment:
Should we perhaps add a new method to `BottomSheetCalculator` to calculate target height:
```swift
       func targetHeight(for contentView: UIView, in superview: UIView, preferredHeight: CGFloat) -> CGFloat {
            if height == .bottomSheetAutomatic {
                let size = contentView.systemLayoutSizeFitting(
                    superview.frame.size,
                    withHorizontalFittingPriority: .required,
                    verticalFittingPriority: .fittingSizeLevel
                )
                return size.height
            } else {
                return height
            }
        }
```

Then we can use it here to get the frame of presented view, and append `handleHeight` in `offset(for:in:height)` func only if the given height is `.bottomSheetAutomatic`.

Reply:
Yep, we can do that